src/algebra/algebra/restrict_scalars.lean

```diff
@@ -148,14 +148,6 @@ add_equiv.refl M
 
 variables [comm_semiring R] [semiring S] [algebra R S] [module S M]
 
-/--
-Note that this lemma relies on the definitional equality `restrict_scalars R S M = M`,
-so usage may result in instance leakage.
-`restrict_scalars.add_equiv_map_smul` is the "hygienic" version.
--/
-lemma restrict_scalars_smul_def (c : R) (x : restrict_scalars R S M) :
-  c • x = ((algebra_map R S c) • x : M) := rfl
```

Comment:
We could keep this as
```suggestion
c • x = (restrict_scalars.add_equiv R S M).symm (algebra_map R S c • restrict_scalars.add_equiv R S M x)
```
if we wanted to, which doesn't abuse the defeq.

Reply:
Good point, I've decided to keep it (renaming from `restrict_scalars_smul_def` to `restrict_scalars.smul_def` to match the lemmas around it).